### PR TITLE
add codesweep false positives list and readme

### PR DESCRIPTION
**Pull request by mdkaraman**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/186345798

# A brief description of the changes

Current behavior: No way of tracking known false positives produced by the AppScan CodeSweep check.

New behavior: 
- File exists where known false positives are tracked
- Security expert (Desire Banse) is codeowner of the false positives list
- readme file is addded explaining the protocol for handling suspected false positives
- PR template updated with section for documenting details of codesweep failures

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

**Got approval from Angus in the "Cross Product Tech Leads" meeting**

### CODE REVIEW

File: appscan_codesweep/CODESWEEP_FALSE_POSITIVES.MD
Comment (saikumar9) [inline appscan_codesweep/CODESWEEP_FALSE_POSITIVES.MD:14]: Add file path for reference.

**Review by desireB — DISMISSED** [review]
Based on the meeting held on 11/13, I am confident that this process will improve the code security. I approved these changes.

**Review by kristinmerbach — DISMISSED** [review]

File: appscan_codesweep/CODESWEEP_FALSE_POSITIVES.MD
Comment (polographer) [inline appscan_codesweep/CODESWEEP_FALSE_POSITIVES.MD:24]: Do you think we can change this phrasing? this is what I see; SQL injection does exist on Mongo, if we leave the phrasing as it is developers can see this as a blanket but in reality, it is no, the linked PR parameters are not dynamic, they are hardcoded as true, so there is no way that we can inject unless we inject on the "distinct", however this are usually just ids and that easy to validate. The bigger issue here is that this is a rake script this  will not receive unsanitized parameters from the web, however codesweep is not aware of that distinction.

can we rephrase to be, 1. This is a backend rake task and will never receive unsanitized parameters, or 1. this query has the "where" parameters harcoded and the limit is always ids, injection is not possible.

Comment (mdkaraman) [reply appscan_codesweep/CODESWEEP_FALSE_POSITIVES.MD:24]: @polographer Yes, totally agree that the language should be improved here.  I used the same language that was present on the original PR since that had been approved, but that language is not technically accurate.  For this type of false positive we should have a more accurate description in this file.

How does this sound?

1. There is no user input being sent to the `.where` query: `.where(:"applicants.is_ia_eligible" => true)`
2. The only parameter being sent to the query is cast to an integer: `assistance_year = ENV['assistance_year'].to_i`
3. The query is called in a script that is run manually and internally.

Comment (polographer) [reply appscan_codesweep/CODESWEEP_FALSE_POSITIVES.MD]: That sounds great! Thanks

**Review by polographer — DISMISSED** [review]

**Review by desireB — DISMISSED** [review]
I have reviewed this PR and the process will improve the code security. I approved these changes.

**Review by kristinmerbach — DISMISSED** [review]

**Review by saikumar9 — DISMISSED** [review]

**Review by polographer — APPROVED** [review]

**Review by saikumar9 — APPROVED** [review]

**Comment by mdkaraman** [comment]
We got unlucky and GitHub Pull Requests was experiencing degraded performance after we got all approvals and merged latest trunk.  The build and deploy steps stalled out and never completed, nor could they be re-triggered manually.  

I had to push a commit to re-trigger all the workflows. 

I've re-requested approvals.  They should be rubber stamped since virtually nothing has changed (I removed a couple extra spaces from the readme so that I could commit and push.)

**Pull request merged**
